docs/core/common/assets/js/api/core/hooks/data.md

```diff
@@ -0,0 +1,103 @@
+## API - `$e.hooks.data`
+*  **Description**: `$e.hooks.data` api is a manager of _DATA_ hooks, allow you to create custom **data manipulation** 
+of *elementor* data model, and create a dependencies, the _hooks_ attached 
+to  `$e.commands`  and each  **hook** being fired after/before/catch a command, that runs by  `$e.run()`
+*  **Location**: *core/common/assets/js/api/core/hooks/data.js*
+*  **Parent**: [`{HooksBase}`](#HooksBase)
+*  **Methods**: Please look at parent: `{HooksBase}` for all the methods.
+* ***Important***: All hooks should be created by extending [`{( $e.modules.hookData )}`](#e-modules-hooks-data) located at: `core/common/assets/js/api/modules/hooks/data/`.
+	
+	| Class                             | Description                                                                                            
+	|-----------------------------------|--------------------------------------------------------------------------------
+	| `$e.modules.hookData.Base`        | Naked base for creating custom data hooks.                   
+	| `$e.modules.hookData.After`       | Used to create a hook, runs after command being executed.
+	| `$e.modules.hookData.Dependency`  | Used to create a hook, runs before command being executed and used as dependency.
+	| `$e.modules.hookData.Catch`       | Used to create a hook, runs when command failed.
+
+*  **Examples**:
+   * Built in hooks:  *`assets/dev/js/editor/document/hooks/data`*
+   * Register data hook that runs **_after_** command runs:
+        ```javascript
+        // Example of data hook, fired after the command runs.
+        // Important: Available to run in the console but depends on $e.components example#1.
+        class CustomDataHook extends $e.modules.hookData.After {
+            getCommand() {
+                // Command to hook.
+                return 'custom-component/example';
+            }
+        
+            getId() {
+                // Unique id for the hook.
+                return 'custom-component-example-data-hook';
+            }
+
+            // Recommended function, used for optimization, if the container type is known in advance,
+            // you can pass it here.
+            //
+            // getContainerType() {
+            // If `args.container.type` is always the same for the hook return it:
+            // return 'container_type';
+            // }
+        
+            /* Optional function, the conditions for hook to be run. */
+            getConditions( args ) {
+                return 'value' === args.property;
+            }
+        
+            /*
+             * The actual hook logic.
+             */
+            apply( args, containers ) {
+                console.log( 'My hook custom logic', 'args: ', args, 'containers: ', containers );
+            }
```

Comment:
We cannot since the hooks mechanism should not know about the existence of container. I changed it to result, thanks for notifying that it means that 'getContainerType' should exist only in base hooks of editor. Since the container exists only in editor. 


if the example were using one of editor commands then it would make sense.